### DISCUSSION TITLE
Is there a way to enable manager (github-actions) for a specific file per baseBranches?

### DISCUSSION THREAD

**Discussion by oddiiie** [renovatebot/renovate#23045] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Hi, folks!!!
Thank you so much for developing.
I would like to know How can I configure the configuration file.

I have 5 GHA workflow files like below.

```bash
$ tree .github
.github
└── workflows
    ├── test.yaml
    ├── master.yaml
    ├── service-a.yaml
    ├── service-b.yaml
    └── service-c.yaml
```

I would like to enable github-actions manager and configure baseBranch for each GHA workflow file like below.
- test.yaml -> disable github-acitons manager
- master.yaml -> baseBranch: master 
- service-a.yaml -> baseBranch: service-a
- service-b.yaml -> baseBranch: service-b
- service-c.yaml -> baseBranch: service-c

### What you tried so far.

I tried matchFiles and matchPaths, but doesn't work.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by viceice** [comment]
add match base branch too to same rules?

**Reply by oddiiie** [reply]
@viceice 
Thank you for your reply!

> add match base branch too to same rules?

If I can.

**Reply by viceice** [reply]
sure

**Reply by oddiiie** [reply]
@viceice 
I tried some configurations but I could not be configured as expected.
 
This configuration is much closer to what was envisioned, but all GHA workflow files have changed in each PR.
What we expect is that only one file is changed for each PR.([PR list](https://github.com/EigoOda/testy/pulls))
- main.yaml -> baseBranch: main
- service-a.yaml -> baseBranch: service-a
- ...
```json5
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base",
     ":prHourlyLimitNone"
   ],
  "enabledManagers": ["github-actions"],
  "labels": ["dependencies"],
  "baseBranches": ["main", "service-a", "service-b", "service-c"],
  "packageRules": [
    {
      "matchBaseBranches": ["main"],
      "matchFiles": ["main.yaml"]
    },
    {
      "matchBaseBranches": ["service-a"],
      "matchFiles": ["service-a.yaml"]
    },
    {
      "matchBaseBranches": ["service-b"],
      "matchFiles": ["service-b.yaml"]
    },
    {
      "matchBaseBranches": ["service-c"],
      "matchFiles": ["service-c.yaml"]
    }
  ]
}
```

Do you have any configuration ideas? Thanks!

**Reply by rarkins** [reply]
Package rules which have only `matchX` params are pointless and do nothing. i.e. all you've done is match, but then there's no ACTION, such as changing group name, or additional branch prefix, or things like that.

In your original description you talk about wanting to _enable_ things, but Renovate is enabled by default. It sounds like you need to _disable_ things?

**Reply by oddiiie** [reply]
Thanks.
With the following settings, we were able to create a PR for baseBranch as expected.
However, I am not quite sure how to disable it, and have created unnecessary [PRs](https://github.com/EigoOda/testy/pull/12).
We would like to avoid creating the bottom four PRs on the [PR list](https://github.com/EigoOda/testy/pulls).

How can I configure json file?

```json5
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base",
     ":prHourlyLimitNone"
   ],
  "enabledManagers": ["github-actions"],
  "labels": ["dependencies"],
  "baseBranches": ["main", "service-a", "service-b", "service-c"],
  "packageRules": [
    {
      "groupName": "main",
      "matchBaseBranches": ["main"],
      "matchPaths": ["**/main.yaml"]
    },
    {
      "groupName": "service-a",
      "matchBaseBranches": ["service-a"],
      "matchPaths": ["**/service-a.yaml"]
    },
    {
      "groupName": "service-b",
      "matchBaseBranches": ["service-b"],
      "matchPaths": ["**/service-b.yaml"]
    },
    {
      "groupName": "service-c",
      "matchBaseBranches": ["service-c"],
      "matchPaths": ["**/service-c.yaml"]
    }
  ]
}
```

**Reply by rarkins** [reply, marked as answer]
You need to disable all then reenable selectively:

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base",
     ":prHourlyLimitNone"
   ],
  "enabledManagers": ["github-actions"],
  "labels": ["dependencies"],
  "baseBranches": ["main", "service-a", "service-b", "service-c"],
  "packageRules": [
    {
      "description": "Disable everything before selectively enabling",
      "matchPackageNames": ["*"],
      "enabled": false
    },
    {
      "enabled": true,
      "groupName": "main",
      "matchBaseBranches": ["main"],
      "matchPaths": ["**/main.yaml"]
    },
    {
      "enabled": true,
      "groupName": "service-a",
      "matchBaseBranches": ["service-a"],
      "matchPaths": ["**/service-a.yaml"]
    },
    {
      "enabled": true,
      "groupName": "service-b",
      "matchBaseBranches": ["service-b"],
      "matchPaths": ["**/service-b.yaml"]
    },
    {
      "enabled": true,
      "groupName": "service-c",
      "matchBaseBranches": ["service-c"],
      "matchPaths": ["**/service-c.yaml"]
    }
  ]
}
```

**Reply by oddiiie** [reply]
Thanks!!!
